### ISSUE TITLE
Multiple Tool Calls in Single Response

### ISSUE THREAD

**Issue by b4syth** [OlympiaAI/raix#6]
First I want to thank you Obie, for publishing this open source Gem.  It is very much appreciated.

I have a question regarding tool calls that have multiple entries. I was using FunctionDispatch and a tool call to the `gpt-4o-mini` model to run a `save_idea` function/tool.  In a single response, I had multiple entries in the tool_calls array, but the `chat_completion` only processed the first tool call.  I see in the [source code](https://github.com/OlympiaAI/raix/blob/1da857bdf020e170022036aa5f372a0ee9e65726/lib/raix/chat_completion.rb#L94) that you have a TODO for parallel calls.  Is this what you mean by parallel calls?  If so, is there a reason you didn't simply iterate over the array and make the call numerous times? e.g.

```ruby
  tool_calls =  response.dig("choices", 0, "message", "tool_calls") || []
  tool_calls.pluck("function").each do |function|
          @current_function = function["name"]
          # dispatch the called function
          arguments = JSON.parse(function["arguments"].presence || "{}")
          arguments[:bot_message] = bot_message if respond_to?(:bot_message)
          return send(function["name"], arguments.with_indifferent_access)
  end
```

Here is an abbreviated example of the response I had:
```json
        {
          "id": "chatcmpl-ANN1RqZCD5BCfqFBwSuWWmJR2QuYv",
          "object": "chat.completion",
          "created": 1730133405,
          "model": "gpt-4o-mini-2024-07-18",
          "choices": [
            {
              "index": 0,
              "message": {
                "role": "assistant",
                "content": "To summarize our conversation... Now, I will save these ideas to the database. Please hold on for a moment!",
                "tool_calls": [
                  {
                    "id": "call_4dyhWn8spP2Fs9svm9HPQv4P",
                    "type": "function",
                    "function": {
                      "name": "save_idea",
                      "arguments": "{\"concept_id\": 581329532, \"name\": \"Target Market\", \"description\": \"Startup technology companies.\"}"
                    }
                  },
                  {
                    "id": "call_Ito6MQ7SKMLViRzdEgEEjXW8",
                    "type": "function",
                    "function": {
                      "name": "save_idea",
                      "arguments": "{\"concept_id\": 993240211, \"name\": \"Job to be Done\", \"description\": \"Helping founders develop a winning strategy for their business.\"}"
                    }
                  },
                  {
                    "id": "call_F2p42pqNpX5PV2xPq28iJ3zb",
                    "type": "function",
                    "function": {
                      "name": "save_idea",
                      "arguments": "{\"concept_id\": 267604924, \"name\": \"Substitute\", \"description\": \"Tools like 'The Lean Canvas' for documenting strategy.\"}"
                    }
                  },
                  ...
                ],
                "refusal": null
              },
              "logprobs": null,
              "finish_reason": "tool_calls"
            }
          ],
          "usage": {
            "prompt_tokens": 1232,
            "completion_tokens": 557,
            "total_tokens": 1789,
            "prompt_tokens_details": {
              "cached_tokens": 1152
            },
            "completion_tokens_details": {
              "reasoning_tokens": 0
            }
          },
          "system_fingerprint": "fp_f59a81427f"
        }
```

**Comment by obie** [comment]
Thanks for opening this issue, but unfortunately just noticing it now. I will investigate the TODO and get back to you asap. It's possible that the solution is very easy.

**Comment by obie** [comment]
Addressed in v0.4.4

Thank you!

**Comment by b4syth** [comment]
So appreciated! 🙏
